instrumentation/jaxrs/jaxrs-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxrsAnnotationsInstrumentation.java

```diff
@@ -67,52 +69,63 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class JaxRsAnnotationsAdvice {
 
-    @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void nameSpan(
-        @Advice.This Object target,
-        @Advice.Origin Method method,
-        @Advice.Local("otelCallDepth") CallDepth callDepth,
-        @Advice.Local("otelHandlerData") HandlerData handlerData,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
-      callDepth = CallDepth.forClass(Path.class);
-      if (callDepth.getAndIncrement() > 0) {
-        return;
+    public static class AdviceScope {
+      private HandlerData handlerData;
+      private final CallDepth callDepth;
+      private Context context;
+      private Scope scope;
+
+      public AdviceScope(CallDepth callDepth) {
```

Comment:
any reason for having a public constructor and the `enter` method instead of just moving the `enter` logic to the constructor? 

Doing the latter would make it impossible to construct `AdviceScope` without having the `enter` logic execute, which I think is what we want? You also did this already for `CxfRequestContextInstrumentation`.

The same applies to other `public AdviceScope` constructors in this PR

Reply:
good point, I've simplified this in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/14168/commits/5776b56073ab2ba061ec5771be04f817693fcaf2